src/ray/raylet/worker_pool.cc

```diff
@@ -656,18 +677,22 @@ void WorkerPool::TryKillingIdleWorkers() {
   }
 
   // Kill idle workers in FIFO order.
-  for (auto it = idle_of_all_languages.begin();
-       it != idle_of_all_languages.end() &&
+  for (auto it = idle_of_all_languages_.begin();
```

Comment:
Use for each syntax and break the loop when the running size is smaller than soft limit?